Makefile

```diff
@@ -0,0 +1,188 @@
+	$(call set_env_var,DAEMON_IMAGE,$*)
+
+do.image.%: daemon.% ;
+
+clean.image.%: do.image.%
+	@$(call set_env_var,STAGING_DIR,$*) rm -rf $(STAGING_DIR)
+
+stage: $(foreach p, $(FLAVORS_TO_BUILD), stage.$(p)) ;
+build: $(foreach p, $(FLAVORS_TO_BUILD), do.image.$(p))
+push:  $(foreach p, $(FLAVORS_TO_BUILD), do.image.$(p))
+
+build.parallel:
+# Due to output-sync, will not output results until finished so there is no text interleaving
+	@$(MAKE) --jobs=4 --output-sync build
+
+
+# ==============================================================================
+# Clean targets
+.PHONY: clean clean.nones clean.all clean.nuke
+
+clean: $(foreach p, $(FLAVORS_TO_BUILD), clean.image.$(p))
+
+clean.nones:
+	@docker rmi -f $(shell docker images | egrep "^<none> " | awk '{print $$3}') || true
+
+clean.all: clean.nones
+	@rm -rf staging-*
+	# Don't mess with other registries for some semblance of a safe nuke.
+	@docker rmi -f \
+		$(shell docker images | egrep "^$(REGISTRY)/daemon(-base)? " | awk '{print $$3}') || true
+
+clean.nuke: clean.all
+	@docker rmi -f \
+		$(shell docker images | egrep "^.*/daemon(-base)? " | awk '{print $$3}') || true
+
+
+# ==============================================================================
+# Test targets
+.PHONY: lint test.staging
+
+lint:
+	flake8
+
+test.staging:
+	DEBUG=1 tests/stage-test/test_staging.sh
+
+# ==============================================================================
+# Help
+.PHONY: help show.flavors flavors.modified
+
+help:
+	@echo ''
+	@echo 'Usage: make [OPTIONS] ... <TARGETS>'
+	@echo ''
+	@echo 'TARGETS:'
+	@echo ''
+	@echo '  Building:'
+	@echo '    stage             Form staging dirs for all images. Dirs are reformed if they exist.'
+	@echo '    build             Build all images. Staging dirs are reformed if they exist.'
+	@echo '    build.parallel    Build all images in parallel.'
+	@echo '    push              Push release images to registry.'
+	@echo ''
+	@echo '  Clean:'
+	@echo '    clean             Remove images and staging dirs for the current flavors.'
+	@echo '    clean.nones       Remove all image artifacts tagged <none>.'
+	@echo '    clean.all         Remove all images and all staging dirs. Implies "clean.nones".'
+	@echo '                      Will only delete images in the specified REGISTRY for safety.'
+	@echo '    clean.nuke        Same as "clean.all" but will not be limited to specified REGISTRY.'
+	@echo '                      USE AT YOUR OWN RISK! This may remove non-project images.'
+	@echo ''
+	@echo '  Testing:'
+	@echo '    lint              Lint the source code.'
+	@echo '    test.staging      Perform stageing integration test.'
+	@echo ''
+	@echo '  Help:'
+	@echo '    help              Print this help message.'
+	@echo '    show.flavors      Show all flavor options to FLAVORS_TO_BUILD.'
+	@echo "    flavors.modified  Show the flavors impacted by this branch's changes vs master."
```

Comment:
It does not only show the flavors impacted, it also builds stages accordingly so we need to capture this in the help.

Reply:
Afterthoughts, I think we should re-build images after each PR merges so we can benefit from distro potential security updates. So I'll change the build system in that sense.